File: jamcha-back/src/main/java/ma/jamcha/jamcha/entities/Comment.java
package ma.jamcha.jamcha.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "comments")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Comment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "article_id")
    private Article article;

    @Column(columnDefinition = "TEXT")
    private String content;

    // Simple user fields - no authentication needed
    @Column(name = "user_email", nullable = false)
    private String userEmail;

    @Column(name = "user_username", nullable = false)
    private String userUsername;

    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parent_id")
    private Comment parent;

    @Builder.Default
    private Boolean isApproved = true;

    @Builder.Default
    private Integer likesCount = 0;

    @Builder.Default
    private Boolean isReported = false;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @PrePersist
    public void prePersist() {
        if (createdAt == null) createdAt = LocalDateTime.now();
        if (updatedAt == null) updatedAt = LocalDateTime.now();
        if (isApproved == null) isApproved = true;
        if (likesCount == null) likesCount = 0;
        if (isReported == null) isReported = false;
    }

    @PreUpdate
    public void preUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: jamcha-client/src/types/comment.ts
// src/types/comment.ts
export interface Comment {
  id: number;
  articleId: number;
  content: string;
  author: {
    id: number;
    name: string;
    email?: string;
    avatar?: string;
  };
  parentId?: number;
  isApproved: boolean;
  likesCount: number;
  isReported: boolean;
  createdAt: string;
  updatedAt: string;
  replies: Comment[];
}

export interface CommentRequest {
  articleId: number;
  content: string;
  parentId?: number;
  userEmail: string;
  userUsername: string;
}